pkg/lwip/contrib/sock/udp/lwip_sock_udp.c

```diff
@@ -121,29 +122,40 @@ ssize_t sock_udp_recv_buf_aux(sock_udp_t *sock, void **data, void **ctx,
     if ((res = lwip_sock_recv(sock->base.conn, timeout, &buf)) < 0) {
         return res;
     }
-    if (remote != NULL) {
+    if ((remote != NULL) ||
+        ((aux != NULL) && IS_USED(MODULE_SOCK_AUX_LOCAL)
+                       && IS_ACTIVE(LWIP_NETBUF_RECVINFO))) {
         /* convert remote */
-        size_t addr_len;
+        size_t addr_len = sizeof(ipv4_addr_t);
+        int family = AF_INET;
         if (NETCONNTYPE_ISIPV6(sock->base.conn->type)) {
             addr_len = sizeof(ipv6_addr_t);
-            remote->family = AF_INET6;
-        }
-        else if (IS_ACTIVE(LWIP_IPV4)) {
-            addr_len = sizeof(ipv4_addr_t);
-            remote->family = AF_INET;
+            family = AF_INET6;
         }
-        else {
+        else if (!IS_ACTIVE(LWIP_IPV4)) {
             netbuf_delete(buf);
             return -EPROTO;
         }
+        if (remote != NULL) {
+            remote->family = family;
 #if LWIP_NETBUF_RECVINFO
-        remote->netif = lwip_sock_bind_addr_to_netif(&buf->toaddr);
+            remote->netif = lwip_sock_bind_addr_to_netif(&buf->toaddr);
 #else
-        remote->netif = SOCK_ADDR_ANY_NETIF;
+            remote->netif = SOCK_ADDR_ANY_NETIF;
 #endif
-        /* copy address */
-        memcpy(&remote->addr, &buf->addr, addr_len);
-        remote->port = buf->port;
+            /* copy address */
+            memcpy(&remote->addr, &buf->addr, addr_len);
+            remote->port = buf->port;
+        }
+#ifdef MODULE_SOCK_AUX_LOCAL
+        if ((aux != NULL) && IS_ACTIVE(LWIP_NETBUF_RECVINFO)
```

Comment:
Should `LWIP_NETBUF_RECVINFO` maybe be modeled in dependency to `MODULE_SOCK_AUX_LOCAL`?

Reply:
That would certainly be more user friendly :-) I'll add it.